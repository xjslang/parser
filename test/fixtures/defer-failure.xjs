function testDeferFailure() {
  console.log('function start')
  
  defer {
    console.log('defer 1 - will fail')
    throw new Error('defer failure')
  }
  
  defer {
    console.log('defer 2 - should still run')
  }
  
  console.log('function end')
}

testDeferFailure()
