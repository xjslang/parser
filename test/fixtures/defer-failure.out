function start
function end
defer 2 - should still run
defer 1 - will fail
Error: defer failure
